src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -51,6 +51,8 @@ public Task Invoke(HttpContext httpContext)
             if (endpoint != null)
             {
                 Log.MatchSkipped(_logger, endpoint);
+
+                // Someone else set the endpoint, we'll let them handle the unsetting.
```

Comment:
This felt right to me. However it does leave the success of setting endpoints in the hands of other middleware developers. Thoughts?